### PR TITLE
Windows: dockerfile tweaks

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

Some small tweaks to Dockerfile.windows to default to Powershell and natural path semantics (can finally be done now all the changes are in master that support this).

### CODE REVIEW

File: Dockerfile.windows [hunk 1]
```diff
@@ -118,30 +118,33 @@
...
+
 # Environment variable notes:
-#  - GO_VERSION must consistent with 'Dockerfile' used by Linux'.
+#  - GO_VERSION must consistent with 'Dockerfile' used by Linux.
 #  - FROM_DOCKERFILE is used for detection of building within a container.
 ENV GO_VERSION=1.7.3 `
     GIT_LOCATION=https://github.com/git-for-windows/git/releases/download/v2.10.1.windows.1/Git-2.10.1-64-bit.exe `
-    GOPATH=C:/go;C:/go/src/github.com/docker/docker/vendor `
+    GOPATH=C:\go;C:\go\src\github.com\docker\docker\vendor `
+	GOROOT=C:\go `
```
Comment (jhowardmsft) [inline Dockerfile.windows]: Tab vs spaces. Corrected.

File: Dockerfile.windows [hunk 2]
```diff
@@ -118,30 +118,33 @@
 # The number of build steps below are explicitly minimised to improve performance.
 FROM microsoft/windowsservercore
 
+# Use PowerShell as the default shell
+SHELL ["powershell", "-command"]
+
 # Environment variable notes:
-#  - GO_VERSION must consistent with 'Dockerfile' used by Linux'.
+#  - GO_VERSION must consistent with 'Dockerfile' used by Linux.
```
Comment (friism) [inline Dockerfile.windows]: missing "be"?

File: Dockerfile.windows [hunk 3]
```diff
@@ -118,30 +118,33 @@
...
+  `
   $ErrorActionPreference = 'Stop'; `
   Function Download-File([string] $source, [string] $target) { `
-   $wc = New-Object net.webclient; $wc.Downloadfile($source, $target) `
+    $wc = New-Object net.webclient; $wc.Downloadfile($source, $target) `
   } `
   `
   Write-Host INFO: Downloading git...; `
-  Download-File %GIT_LOCATION% gitsetup.exe; `
+  Download-File $env:GIT_LOCATION gitsetup.exe; `
```
Comment (friism) [inline Dockerfile.windows]: Since this is PowerShell, I think the canonical casing is `Env`, but not very important

**Review by friism — COMMENTED** [review]
some small nits, lgtm though

File: Dockerfile.windows [hunk 4]
```diff
@@ -160,18 +163,18 @@ RUN `
   New-ItemProperty $installKey -Name 'Inno Setup CodeFile: Path Option' -Value 'CmdTools' -PropertyType 'String' -Force; `
   New-ItemProperty $installKey -Name 'Inno Setup CodeFile: Bash Terminal Option' -Value 'ConHost' -PropertyType 'String' -Force; `
   New-ItemProperty $installKey -Name 'Inno Setup CodeFile: CRLF Option' -Value 'CRLFCommitAsIs' -PropertyType 'String' -Force; `
-  Start-Process gitsetup.exe -ArgumentList '/VERYSILENT /SUPPRESSMSGBOXES /CLOSEAPPLICATIONS /DIR=c:\git\' -Wait; `
+  Start-Process gitsetup.exe -ArgumentList '/VERYSILENT /SUPPRESSMSGBOXES /CLOSEAPPLICATIONS /DIR=C:\git\' -Wait; `
   `
   Write-Host INFO: Expanding go..."; `
```
Comment (jhowardmsft) [inline Dockerfile.windows]: fixed

File: Dockerfile.windows (hunk 3 above)
Comment (jhowardmsft) [inline Dockerfile.windows]: fixed

File: Dockerfile.windows (hunk 2 above)
Comment (jhowardmsft) [inline Dockerfile.windows]: fixed

**Comment by jhowardmsft** [comment]
OK to move forward?

**Review by thaJeztah — APPROVED** [review]
looks like the comments were addressed, so let me

LGTM

**Comment by jhowardmsft** [comment]
@tiborvass @vdemeester PTAL 😸

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**